mlrun/model_monitoring/db/tsdb/v3io/v3io_connector.py

```diff
@@ -0,0 +1,407 @@
+        )
+
+        def apply_storey_filter():
+            graph.add_step(
+                "storey.Filter",
+                "FilterNotNone",
+                after="FilterAndUnpackKeys3",
+                _fn="(event is not None)",
+            )
+
+        apply_storey_filter()
+        apply_tsdb_target(name="tsdb3", after="FilterNotNone")
+
+    def write_application_event(
+        self,
+        event: dict,
+        kind: mm_constants.WriterEventKind = mm_constants.WriterEventKind.RESULT,
+    ):
+        """
+        Write a single result or metric to TSDB.
+        """
+
+        event[mm_constants.WriterEvent.END_INFER_TIME] = (
+            datetime.datetime.fromisoformat(
+                event[mm_constants.WriterEvent.END_INFER_TIME]
+            )
+        )
+
+        if kind == mm_constants.WriterEventKind.METRIC:
+            # TODO : Implement the logic for writing metrics to V3IO TSDB
+            return
+
+        del event[mm_constants.ResultData.RESULT_EXTRA_DATA]
+        try:
+            self._frames_client.write(
+                backend=_TSDB_BE,
+                table=self.tables[mm_constants.MonitoringTSDBTables.APP_RESULTS],
+                dfs=pd.DataFrame.from_records([event]),
+                index_cols=[
+                    mm_constants.WriterEvent.END_INFER_TIME,
+                    mm_constants.WriterEvent.ENDPOINT_ID,
+                    mm_constants.WriterEvent.APPLICATION_NAME,
+                    mm_constants.ResultData.RESULT_NAME,
+                ],
+            )
+            logger.info(
+                "Updated V3IO TSDB successfully",
+                table=self.tables[mm_constants.MonitoringTSDBTables.APP_RESULTS],
+            )
+        except v3io_frames.errors.Error as err:
+            logger.warn(
+                "Could not write drift measures to TSDB",
+                err=err,
+                table=self.tables[mm_constants.MonitoringTSDBTables.APP_RESULTS],
+                event=event,
+            )
+
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Failed to write application result to TSDB: {err}"
+            )
+
+    def delete_tsdb_resources(self, table: str = None):
+        if table:
+            # Delete a specific table
+            tables = [table]
+        else:
+            # Delete all tables
+            tables = mm_constants.MonitoringTSDBTables.list()
+        for table in tables:
+            try:
+                self._frames_client.delete(
+                    backend=mlrun.common.schemas.model_monitoring.TimeSeriesConnector.TSDB,
+                    table=table,
+                )
+            except v3io_frames.errors.DeleteError as e:
+                if "No TSDB schema file found" not in str(e):
+                    logger.warning(
+                        f"Failed to delete TSDB table '{table}'",
+                        err=mlrun.errors.err_to_str(e),
+                    )
```

Comment:
And if it's not, we just suppress the error entirely?

Reply:
agree, it's an old `if` that seems quite redundant here. We need to log the err either way. I'll update it.